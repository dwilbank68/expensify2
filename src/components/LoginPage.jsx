import React, { Component } from 'react';
import { connect } from 'react-redux';
import {Link} from 'react-router-dom';

import {startEmailLogin, startGoogleLogin} from '../actions/authActionGenerators.js';

const styles = {
    box: {
        borderRadius: '15px',
        background: 'rgba(255,255,255,.9)'
    }
}

export
class LoginPage extends Component {

    constructor(props, context){
        super(props, context);
        this.state = {
            email:'',
            password:''
        }
       this.onSubmit = this.onSubmit.bind(this)
    }

    onEmailChange = e => {
        const email = e.target.value;
        this.setState(() => ({email}));
    }

    onPasswordChange = e => {
        const password = e.target.value;
        this.setState(() => ({password}));
    }

    onSubmit = e => {
        e.preventDefault();
        const {email, password} = this.state;
        if (!email || !password) {
            this.setState( () => ({error: 'please provide email and password'}));
        } else {
            this.setState( () => ({error: ''}));
            this.props.startEmailLogin( email, password )
        }
    }

    render() {
        const {startGoogleLogin} = this.props;
        return (
            <div className="log-in boxed-view">
                <div    className="boxed-view__box"
                        style={styles.box}>
                    <h1>Login</h1>

                    <button onClick={startGoogleLogin}>
                        Login with Google
                    </button>

                    {/*{this.state.error ? <p>{this.state.error}</p> : undefined}*/}

                    <form onSubmit ={this.onSubmit} noValidate className="boxed-view__form">
                        <input type="email"
                               value={this.state.email}
                               onChange={this.onEmailChange}
                               placeholder="Email"/>
                        <input type="password"
                               value={this.state.password}
                               onChange={this.onPasswordChange}
                               placeholder="Password"/>
                        <button className="button">
                            Log In
                        </button>
                    </form>

                    <Link to="/signup">Need an account?</Link>
                </div>
            </div>

            //     {/*<button onClick={startEmailLogin}>*/}
            //         {/*Login with Email*/}
            //     {/*</button>*/}
            // {/*</div>*/}
        );
    }
}

const mapDispatchToProps = dispatch => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startEmailLogin: (email, password) => dispatch(startEmailLogin(email, password)),
})

export default connect(null, mapDispatchToProps)(LoginPage);
