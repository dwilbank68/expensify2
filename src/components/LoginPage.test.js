import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from './LoginPage.jsx';

let wrapper, startGoogleLogin;

beforeEach(() => {
    startGoogleLogin = jest.fn();
    wrapper = shallow(
        <LoginPage startGoogleLogin={startGoogleLogin}/>
    );
})

test('should render LoginPage', () => {
    expect(wrapper).toMatchSnapshot();
})

test('should call startGoogleLogin on button click', () => {
    wrapper
        .find('button').at(0)
        .simulate('click')
    expect(startGoogleLogin).toHaveBeenCalled();
})