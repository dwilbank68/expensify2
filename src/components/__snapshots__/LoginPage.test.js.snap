// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage 1`] = `
<div
  className="log-in boxed-view"
>
  <div
    className="boxed-view__box"
    style={
      Object {
        "background": "rgba(255,255,255,.9)",
        "borderRadius": "15px",
      }
    }
  >
    <h1>
      Login
    </h1>
    <button
      onClick={[MockFunction]}
    >
      Login with Google
    </button>
    <form
      className="boxed-view__form"
      noValidate={true}
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        className="button"
      >
        Log In
      </button>
    </form>
    <Link
      replace={false}
      to="/signup"
    >
      Need an account?
    </Link>
  </div>
</div>
`;
